From 16e08f5107056cce98e52d2455aca9c747d88f71 Mon Sep 17 00:00:00 2001
From: Feng Zhang <feng@radxa.com>
Date: Wed, 19 Feb 2025 16:38:02 +0800
Subject: [PATCH] drivers: mtd: Fix boot issues caused by SPI NOR

This is a temporary patch to fix the boot issue caused by SPI FLASH driver
problems, which prevents boards with SPI NOR from booting.
---
 drivers/mtd/mtdpart.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/mtd/mtdpart.c b/drivers/mtd/mtdpart.c
index 357661b62..5a3f3abd3 100644
--- a/drivers/mtd/mtdpart.c
+++ b/drivers/mtd/mtdpart.c
@@ -673,6 +673,7 @@ int parse_mtd_partitions(struct mtd_info *master, const char *const *types,
 			ret = mtd_part_do_parse(parser, master, &pparts, data);
 			if (ret <= 0)
 				mtd_part_parser_put(parser);
+			return 0;
 		}
 		/* Found partitions! */
 		if (ret > 0) {
-- 
2.25.1

